### PR TITLE
Add com.getpostman.Postman

### PR DESCRIPTION
**Pull request by ghisvail**

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.getpostman.Postman

**Comment by nedrichards** [comment]
Reading their EULA - I think this has to be extra-data rather than copied into the flatpak and distributed directly. https://www.getpostman.com/licenses/postman_eula

**Comment by ghisvail** [comment]
@nedrichards I don't follow. Could you be more explicit please?

**Comment by nedrichards** [comment]
> "Postman grants You a revocable, non-exclusive, non-transferable, non-sublicenseable, license to Use"

This means, like Slack - that we can't distribute the binaries directly. We can only link to them in the flatpak using extra-data and the user downloads them, runs instructions and scripts provided in the flatpak and they then run from inside the flatpak container. So instead of a 'file' type it should be 'extra-data'

**Comment by ghisvail** [comment]
@a85 @prashantagarwal @preethammavin could any of you confirm? ^

**Comment by bilelmoussaoui** [comment]
The desktop file could use some love too, it lacks the categories

**Comment by nedrichards** [comment]
Yes - we need to add that to the requirements actually. I'm seeing more and more without them.

**Comment by ghisvail** [comment]
It matches the minimal desktop file currently defined in the [flatpak docs](http://docs.flatpak.org/en/latest/conventions.html#desktop-files). If you want less of these, perhaps adjusting the docs would be a good place to start imo.

**Comment by ghisvail** [comment]
Categories added.

**Comment by nedrichards** [comment]
@ghisvail fair point - I'll file an issue to do that.

File: com.getpostman.Postman.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - install -Dm755 run.sh /app/bin/Postman
+      - install -Dm644 com.getpostman.Postman.desktop /app/share/applications/com.getpostman.Postman.desktop
+      - install -Dm644 com.getpostman.Postman.appdata.xml /app/share/metainfo/com.getpostman.Postman.appdata.xml
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
+        url: https://dl.pstmn.io/download/version/6.1.3/linux64
+        sha256: 6e08dae1524416396921337dd785e84436c1887a7d835b24c69ea68dce13290d
+        dest-filename: Postman-linux-x64-6.1.3.tar.gz
```
Comment (eszlari) [inline com.getpostman.Postman.yml]: You could give this a generic name (e.g. `dest-filename: Postman.tar.gz`), so you don't have to update that too (and remove the glob above).

File: com.getpostman.Postman.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+modules:
+  - name: Postman
+    buildsystem: simple
+    build-commands:
+      - tar -xf Postman-*.tar.gz
+      - mkdir -p /app/Postman
+      - mv Postman/app/* /app/Postman
+      - install -Dm644 /app/Postman/resources/app/assets/icon.png /app/share/icons/hicolor/128x128/apps/com.getpostman.Postman.png
+      - install -Dm755 run.sh /app/bin/Postman
+      - install -Dm644 com.getpostman.Postman.desktop /app/share/applications/com.getpostman.Postman.desktop
```
Comment (eszlari) [inline com.getpostman.Postman.yml]: `install -Dm644 -t /app/share/applications com.getpostman.Postman.desktop`

File: com.getpostman.Postman.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  - --filesystem=home
+modules:
+  - name: Postman
+    buildsystem: simple
+    build-commands:
+      - tar -xf Postman-*.tar.gz
+      - mkdir -p /app/Postman
+      - mv Postman/app/* /app/Postman
+      - install -Dm644 /app/Postman/resources/app/assets/icon.png /app/share/icons/hicolor/128x128/apps/com.getpostman.Postman.png
+      - install -Dm755 run.sh /app/bin/Postman
```
Comment (eszlari) [inline com.getpostman.Postman.yml]: `m755` could be dropped

File: com.getpostman.Postman.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - type: file
+        only-arches:
+          - i386
+        url: https://dl.pstmn.io/download/version/6.1.3/linux32
+        sha256: 8831faff3001a921a3ab9894e0cbc966ecc2c91b5186edb8cfcf84fd5ef6844d
+        dest-filename: Postman-linux-ia32-6.1.3.tar.gz
+      - type: script
+        dest-filename: run.sh
+        commands:
+          - exec env TMPDIR=$XDG_CACHE_HOME /app/Postman/Postman $@
```
Comment (eszlari) [inline com.getpostman.Postman.yml]: quotes: `"$@"`

File: com.getpostman.Postman.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    </p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://www.getpostman.com/img/v2/postman/Postman-most-complete-toolchain.png</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">https://www.getpostman.com</url>
+  <releases>
+    <release version="6.1.3" date="2018-05-25"/>
```
Comment (eszlari) [inline com.getpostman.Postman.appdata.xml]: I would put `<releases>` at the end of the file (easy to find, maybe you want to maintain a list).

**Comment by eszlari** [comment]
Have you considered YAML?

File: com.getpostman.Postman.yml (hunk 1 above)
Comment (eszlari) [reply com.getpostman.Postman.yml]: Why do you even use `type: file` and not `type: archive`?

File: com.getpostman.Postman.appdata.xml (hunk 5 above)
Comment (ghisvail) [reply com.getpostman.Postman.appdata.xml]: I usually don't.

**Comment by ghisvail** [comment]
Is there a YAML version for AppData?

File: com.getpostman.Postman.yml (hunk 1 above)
Comment (ghisvail) [reply com.getpostman.Postman.yml]: Try with `archive` and you'll see. It fails on my machine.

**Comment by eszlari** [comment]
> Is there a YAML version for AppData?

I mean JSON -> YAML.

**Comment by ghisvail** [comment]
> Was looking at the wrong repo.

I was about to suggest a mixup. That's the first submission I am doing with YAML. I was using JSON for my previous ones and will probably convert them at some point.

File: com.getpostman.Postman.yml (hunk 1 above)
Comment (eszlari) [reply com.getpostman.Postman.yml]: https://github.com/flatpak/flatpak-builder/issues/158

**Comment by eszlari** [comment]
Maybe a minor thing, but I think *.yaml is more common then *.yml as a file extension.

**Comment by ghisvail** [comment]
That's personal preference at the end of the day. I followed the official [post](https://blogs.gnome.org/alexl/2018/05/04/yamling-the-flathub/) from @alexlarsson and he's using .yml for his [example](https://github.com/flathub/org.gnome.eog/commit/d8f2dda8841031e326b1e7ac97367286f1a652c6).

Unless there are more authoritative instructions somewhere that say otherwise?

**Comment by ghisvail** [comment]
@nedrichards Could you provide further information about the Slack or Skype way of installing via Flatpak. I have looked at both manifests, but I fail to understand how it works in the end.

From a naive first look, I can see the content of the .deb is moved to some `export/` folder, but the wrapper script calls something under `/app/extra/bin`. I can't find the connection between both.

**Comment by nedrichards** [comment]
@ghisvail it was easier for me to just quickly convert it than explain properly (sorry, I'm a bit tired) but hopefully it makes a bit more sense now. Basically you need to:

1) Have enough stuff installed up front that your application will show up in software stores and has enough about it to unpack the archives it downloads and run the commands (this is why I added the icon to the distribution, otherwise you wouldn't show up to be installed since the archive is only downloaded after the decision to install is made)
2) Have an apply_extra script (which is like build-commands, but for extra-data) and at least one extra-data source.
3) When you try and install the app it downloads the extra-data part and runs the apply_extra script. the reason for the /app/extra path is that this bit reserved as writable in the otherwise immutable flatpak
4) You have an otherwise perfectly normal flatpak with all the advantages and disadvantages you'd get before. The only real disadvantage of this approach vs your previous one is that users don't get diff updates because the archive isn't retained, they have to download a new extra-data payload with each update.

**Comment by nedrichards** [comment]
bot, build com.getpostman.Postman

**Comment by ghisvail** [comment]
@nedrichards You did well. Thanks for providing further details afterwards. It's becoming clearer.

We might also want to inform somewhere that this is not an installation method officially endorsed by upstream? Did you do something similar with Slack and Skype?

**Comment by nedrichards** [comment]
Yes - I think that's best. You should add that to the description in the appdata. But once that's done - and you're happy that everything works and is up to date (I tested briefly, but not in detail) then this is good to merge.

**Comment by ghisvail** [comment]
> You should add that to the description in the appdata.

Done.

> then this is good to merge.

I'll take this opportunity to thank for your review, patience and mentorship.

Very much appreciated.

**Comment by bilelmoussaoui** [comment]
Shouldn't the appdata proposed to upstream? I have already requested a .desktop and an .appdata file in the past and Postman team seems to be willing to add those.

**Comment by ghisvail** [comment]
I'll make sure to contact them once this is validated and merged.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.getpostman.Postman

**Comment by ghisvail** [comment]
Looks like the build is stuck for some reason?

**Comment by nedrichards** [comment]
Built fine: https://flathub.org/builds/#/builders/2/builds/2419 I can see it from the CLI too. Will take a peek into GNOME Software etc.

**Comment by nedrichards** [comment]
Hrm yeah, I can see it fine in GNOME Software - so I suspect there's either an issue with the web front end or it's being a bit slow in updating.

**Comment by nedrichards** [comment]
Yeah, visible now https://flathub.org/apps/details/com.getpostman.Postman

**Pull request closed without merging**